### ISSUE TITLE
Support for Multiple Screens?

### ISSUE THREAD

**Issue by nkymut** [niw/WindowAlignment#1]
Thank you for releasing such a useful software.
I appreciate the software allows me to precisely set the screen's position and size numerically and customise its key combinations.

Is the software currently supporting multiple displays? If so, how should I specify the position and size for the secondary display?
When using multiple displays, specifying values like screen.width, screen.height, etc. behaves unpredictably on the secondary screen right now. 

here's my current setting. 

```json
{
  "actions" : [
      {
        "hot_key" : {
          "key_code" : 124,
          "modifiers" : [
            "shift",
            "command"
          ]
        },
        "move" : {
          "x" : "screen.width * 0.5",
          "y" : "0"
        },
        "resize" : {
          "width" : "screen.width * 0.5",
          "height" : "screen.height"
        }
      },
      {
        "hot_key" : {
          "key_code" : 123,
          "modifiers" : [
            "shift",
            "command"
          ]
        },
        "move" : {
          "x" : "0",
          "y" : "0"
        },
        "resize" : {
          "width" : "screen.width * 0.5",
          "height" : "screen.height"
        }
      },
      
      
    {
      "hot_key" : {
        "key_code" : 126,
        "modifiers" : [
          "shift",
          "command"
        ]
      },
      "move" : {
        "x" : "window.x",
        "y" : "0"
      },
      "resize" : {
        "width" : "screen.width * 0.5",
        "height" : "screen.height * 0.5"
      }
    },
    {
      "hot_key" : {
        "key_code" : 125,
        "modifiers" : [
          "shift",
          "command"
        ]
      },
      "move" : {
        "x" : "window.x",
        "y" : "screen.height * 0.5"
      },
      "resize" : {
        "width" : "screen.width * 0.5",
        "height" : "screen.height * 0.5"
      }
    }
  ]
}

```

**Comment by niw** [comment]
Current implementation is assuming that each screen ("Space") is assigned to each display (a physical display monitor) by enabling "Displays have separate Spaces" on "Desktop & Dock", also disabling "Stage Manager" (which are default settings of macOS,) and thus, each window has its screen always (which is switching when you drag the window to the other screen, which is a Space on another display,) and currently only within that screen we can align windows.

My best guess is that your demand is to add ability to specficy screen for aligning or moving window between screens. Is that correct? If so, it needs extra implementation to do so.

**Comment by nkymut** [comment]
Thank you for the clear explanation!

So, `screen.x` and `screen.y` refer to the origin coordinates of the current screen ("Space") where the window is located at the moment.

With the original settings, 

```JSON
        "move" : {
          "x" : "screen.width * 0.5",
          "y" : "0"
        },
```
the window would jump to an awkward position in the adjacent screen in a multi-screen environment (Screen A and Screen B next to each other), which was problematic.
However, by updating to:

```JSON
        "move" : {
          "x" : "screen.x + screen.width * 0.5",
          "y" : "0"
        },
```
solved the issue and the window now moves to the midpoint of the current screen as expected.

Thank you!

**Comment by niw** [comment]
Ah, yes, you need to use `screen.x` and `screen.y` which can be non-zero value (even negative value, because each screen is placed in the global coordinate where the origin is primary screen origin.)

**Comment by niw** [comment]
Actually I found a bug, that the coordinate conversion is wrong, and now it should be fixed by 8ad75fcc39494ee8b837a4745b1adfc2c9eb2bab.
Also released [0.2.0](https://github.com/niw/WindowAlignment/releases/tag/0.2.0) which includes the change.

**Comment by niw** [comment]
It's open for a while, I believe the problem has been resolved therefore going to close. If it's not, please reopen this or file new issue.
